units/sdlthread.inc

```diff
@@ -66,27 +72,55 @@ type
 {$DEFINE SDL_PASSED_BEGINTHREAD_ENDTHREAD}
 
 type
-  {$IFNDEF DELPHI16UP}
-    TThreadID = Cardinal;
+  { SDL2-for-Pascal: #todo : Needed? }
+  {$IFNDEF FPC}
+    {$IFNDEF DELPHI16UP}
+      TThreadID = Cardinal;
+    {$ENDIF}
   {$ENDIF}
 
-  TpfnSDL_CurrentBeginThread = function(SecurityAttributes: Pointer; StackSize: LongWord; ThreadFunc: TThreadFunc; Parameter: Pointer; CreationFlags: LongWord; var ThreadId: TThreadID): cint;
+  { SDL2-for-Pascal #todo : Explanation needed }
+  TpfnSDL_CurrentBeginThread = function(
+    SecurityAttributes: Pointer; StackSize: LongWord; ThreadFunc: TThreadFunc;
+    Parameter: Pointer {arg}; CreationFlags: LongWord; var ThreadId: TThreadID {threadID}): cuintptr_t; cdecl;
+
+  TpfnSDL_CurrentEndThread = procedure(code: cuint); cdecl;
 
-  TpfnSDL_CurrentEndThread = procedure(ExitCode: cint);
 
-  {**
-   *  Create a thread.
-   *}
 function SDL_CreateThread(fn: TSDL_ThreadFunction; name: PAnsiChar;
-  data: Pointer; pfnBeginThread: TpfnSDL_CurrentBeginThread; pfnEndThread: TpfnSDL_CurrentEndThread): PSDL_Thread; cdecl; overload;
-  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_CreateThread' {$ENDIF} {$ENDIF};
+  data: Pointer;
+  pfnBeginThread: TpfnSDL_CurrentBeginThread;
+  pfnEndThread: TpfnSDL_CurrentEndThread): PSDL_Thread; cdecl; overload;
+  external SDL_LibName;
 
-  {**
-   *  Create a thread.
-   *}
+function SDL_CreateThreadWithStackSize(fn: TSDL_ThreadFunction;
+  name: PAnsiChar; const stacksize: csize_t; data: Pointer;
+  pfnBeginThread: TpfnSDL_CurrentBeginThread;
+  pfnEndThread: TpfnSDL_CurrentEndThread): PSDL_Thread; cdecl; overload;
+  external SDL_LibName;
+
+{ SDL2-For-Pascal:  #note : In the C header are two versions
+    of these macro functions. One for SDL's dynamic API and one without.
+    We can go with this for the moment. Improvement surely possible. }
```

Comment:
Looking at SDL's source code, the library by default is built in the "dynamic API" mode, and building without it requires manually editing the source. Based on that, I think we should translate the "dynamic API" versions of `SDL_CreateThread()` and `SDL_CreateThreadWithStackSize()`. Since said macros call `SDL_CreateThread_REAL()` and `SDL_CreateThreadWithStackSize_REAL()`, we could avoid having to overload the identifiers.